crates/zkvm/zkvm/src/lib.rs

```diff
@@ -204,3 +204,102 @@ impl AggregationInput {
         &self.vk
     }
 }
+
+/// A trait representing a Zero-Knowledge Virtual Machine (ZkVM) interface.
+/// Provides methods for reading inputs, committing outputs, and verifying proofs
+/// within the ZkVM environment.
+pub trait ZkVm {
+    /// Reads a serialized object from the guest code, deserializing it using Serde.
+    ///
+    /// The input is expected to be written with [`ZKVMInputBuilder::write`].
+    fn read<T: DeserializeOwned>(&self) -> T;
+
+    /// Reads a serialized byte slice from the guest code.
+    ///
+    /// The input is expected to be written with [`ZKVMInputBuilder::write_serialized`].
+    fn read_slice(&self) -> Vec<u8>;
+
+    /// Commits a Serde-serializable object to the public values stream.
+    ///
+    /// Values that are committed can be proven as public parameters.
+    fn commit<T: Serialize>(&self, output: &T);
+
+    /// Commits a pre-serialized byte slice to the public values stream.
+    ///
+    /// This method is intended for cases where the data has already been serialized
+    /// outside of the ZkVM's standard serialization methods. It allows you to provide
+    /// serialized outputs directly, bypassing any further serialization.
+    fn commit_slice(&self, raw_output: &[u8]);
```

Comment:
This is fine as-is, although maybe if you change the other one it also makes sense to rename to `commit_buf`.